addons/delivery/views/sale_order_views.xml

```diff
@@ -40,7 +40,7 @@
                 <field name="is_delivery" column_invisible="True"/>
             </xpath>
             <xpath expr="//field[@name='order_line']/tree" position="attributes">
-                <attribute name="decoration-warning">recompute_delivery_price and is_delivery</attribute>
+                <attribute name="decoration-warning">has_archived_product or (recompute_delivery_price and is_delivery)</attribute>
```

Comment:
```suggestion
                <attribute name="decoration-warning" add="(recompute_delivery_price and is_delivery)" separator="or"/>
```
You should avoid overwriting xml attribute values as it won't work well if multiple inherited views do the same.
Use the attribute manipulation API instead ;).